packages/ListBox/src/helpers/DOMAttributes.js

```diff
@@ -0,0 +1,41 @@
+import uuidv4 from "uuid/v4";
+
+const listBoxId = `listBoxId_${uuidv4()}`;
+
+export function getDOMAttributesForListBoxContainer() {
+  return { tabIndex: "-1" };
+}
+
+export function getDOMAttributesForListBox(state) {
+  // NOTE: replace this with L10n component
+  let activedescendant = "none";
+  const { activeOption } = state;
+
+  if (activeOption && activeOption < Object.keys(state.options).length) {
+    const { id } = state.options[activeOption];
+    if (id) {
+      activedescendant = id;
+    }
+  }
+
+  return {
+    "aria-activedescendant": activedescendant,
+    "aria-labelledby": "listbox-value",
+    id: listBoxId,
+    role: "listbox",
+  };
+}
+
+export const getA11yAttributesForOption = isSelected => {
+  const a11yIsSelected = { "aria-selected": `${isSelected ? "true" : "false"}` };
+  return { role: "option", ...a11yIsSelected };
+};
+
+export function getDOMAttributesForListBoxButton() {
+  return {
+    "aria-haspopup": "popup-picker",
```

Comment:
`aria-haspopup` should be a boolean.  Maybe you want to use `aria-controls` here too?

Reply:
Screen-reader is a whole other feature in this component, that's why I didn't even bother in this PR, I focused on keyboard accessibility specifically. 

Though I put some function in place and for this specifically and can support the implementation later. I have created an issue to address ListBox screen reader support. 

https://github.com/acl-services/paprika/issues/89